File: TopCoder/src/main/java/com/leetcode/String/UniqueMorseCodeWords.java
package com.leetcode.String;

import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

public class UniqueMorseCodeWords {

    public int uniqueMorseRepresentations(String[] words) {
        String[] codes = {
                ".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."
        };

        Set<String> set = Arrays.stream(words).map(word -> {
            StringBuilder stringBuilder = new StringBuilder(word.length());
            for (int i = 0 ; i < word.length() ; i++) {
                int codeIndex = word.charAt(i) - 'a';
                stringBuilder.append(codes[codeIndex]);
            }
            return stringBuilder.toString();
        }).collect(Collectors.toSet());
        return set.size();
    }

    public static void main(String[] args) {
        String[] arr = {"gin", "zen", "gig", "msg"};
        int result = new UniqueMorseCodeWords().uniqueMorseRepresentations(arr);
        System.out.println(result);
    }
}

File: TopCoder/python/misc/UniqueMorseCodeWords.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        codes = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
        resultcodes = []
        for word in words:
            result = ''
            for c in word:
                index = ord(c) - ord('a')
                result += codes[index]
            resultcodes.append(result)
        return len(set(resultcodes))


if __name__ == "__main__":
    sol = Solution()
    words = ["gin", "zen", "gig", "msg"]
    res = sol.uniqueMorseRepresentations(words)
    print(res)